### PR TITLE
🤔 Remove compile warnings and unify for-loop syntax.

### PR DESCRIPTION
**Pull request by destogl**

### CODE REVIEW

**Review by pac48 — COMMENTED** [review]
I compiled the code and ran the tests. This change does remove the warning as intended.  However, I had to change all `std::static_cast` to `static_cast` to get it to compile. Perhaps my gcc/g++ is newer. I'm on version 11.3.0. Otherwise it looks good.

File: kinematics_interface_kdl/src/kinematics_interface_kdl.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ bool KinematicsInterfaceKDL::verify_link_name(const std::string & link_name)
 
 bool KinematicsInterfaceKDL::verify_joint_vector(const Eigen::VectorXd & joint_vector)
 {
-  if (joint_vector.size() != num_joints_)
+  if (std::static_cast<size_t>(joint_vector.size()) != num_joints_)
```
Comment (pac48) [inline kinematics_interface_kdl/src/kinematics_interface_kdl.cpp:198]:
```suggestion
  if (static_cast<size_t>(joint_vector.size()) != num_joints_)
```

File: kinematics_interface_kdl/test/test_kinematics_interface_kdl.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ TEST_F(TestKDLPlugin, KDL_plugin_function)
     ik_->convert_joint_deltas_to_cartesian_deltas(pos, delta_theta, end_effector_, delta_x_est));
 
   // Ensure kinematics math is correct
-  for (auto i = 0l; i < delta_x.size(); i++)
+  for (size_t i = 0; i < std::static_cast<size_t>(delta_x.size()); ++i)
```
Comment (pac48) [inline kinematics_interface_kdl/test/test_kinematics_interface_kdl.cpp:95]:
```suggestion
  for (size_t i = 0; i < static_cast<size_t>(delta_x.size()); ++i)
```

File: kinematics_interface_kdl/test/test_kinematics_interface_kdl.cpp [hunk 3]
```diff
@@ -125,7 +125,7 @@ TEST_F(TestKDLPlugin, KDL_plugin_function_std_vector)
...
```
Comment (pac48) [inline kinematics_interface_kdl/test/test_kinematics_interface_kdl.cpp:128]:
```suggestion
  for (size_t i = 0; i < static_cast<size_t>(delta_x.size()); ++i)
```

**Review by bmagyar — APPROVED** [review]

**Pull request merged**